### PR TITLE
Live TV button navigations

### PR DESCRIPTION
**Pull request by MrChip53**
**Changes**
Change channels with number buttons on remote
Jump/Last button goes to previous channel
Channel up/down work as they should also

**Todo**
- [x] Channel up and down buttons
- [x] Last channel button
- [x] Sort channels by channel number if they arent already

**Issues**

Relevant issue: #103

### CODE REVIEW

File: app/src/main/java/org/jellyfin/androidtv/ui/playback/CustomPlaybackOverlayFragment.java [hunk 1]
```diff
@@ -620,6 +688,75 @@ public boolean onKey(View v, int keyCode, KeyEvent event) {
         }
     };
 
+    private void switchChannelByNumber(String number) {
+        List<ChannelInfoDto> channels = TvManager.getAllChannels();
+        for (ChannelInfoDto channel:
+                channels) {
+            if (channel.getNumber().equals(number)) {
```
Comment (thornbill) [inline app/src/main/java/org/jellyfin/androidtv/ui/playback/CustomPlaybackOverlayFragment.java:695]: A NullPointerException is thrown here if an invalid channel number is entered.

File: app/src/main/java/org/jellyfin/androidtv/util/KeyUtils.kt [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+object KeyUtils {
+	@JvmStatic
+	fun isSelect(code: Int) = code == KeyEvent.KEYCODE_DPAD_CENTER || code == KeyEvent.KEYCODE_ENTER
+
+	@JvmStatic
+	fun isNumber(code: Int) = code >= KeyEvent.KEYCODE_0 && code <= KeyEvent.KEYCODE_9
+
+	@JvmStatic
+	fun isChannelSurfUp(code: Int) = code == KeyEvent.KEYCODE_PAGE_UP || code == KeyEvent.KEYCODE_CHANNEL_UP
+	@JvmStatic
```
Comment (thornbill) [inline app/src/main/java/org/jellyfin/androidtv/util/KeyUtils.kt:14]: Spacing is inconsistent in this file. I would prefer a blank line before each annotated function like was done above.

**Review by thornbill — CHANGES_REQUESTED** [review]
Looks good overall. I couldn't seem to find a way to emulate the paging behavior in the emulator. I did notice entering an incorrect channel number caused an NPE. This also don't seem to be allowing "." to be entered for digital OTA channels like "3.1". I'm not sure how we should account for those channels since remotes may not have a "." button either.

**Comment by MrChip53** [comment]
Okay im going to move this back to draft and try some things out to address the 3.1, 3.2, etc issue. I think it may require a decent rework of the channel up/down logic.

**Comment by PrplHaz4** [comment]
> 
> 
> Looks good overall. I couldn't seem to find a way to emulate the paging behavior in the emulator. I did notice entering an incorrect channel number caused an NPE. This also don't seem to be allowing "." to be entered for digital OTA channels like "3.1". I'm not sure how we should account for those channels since remotes may not have a "." button either.

Maybe this could be handled like a currency input with an applied `.` that always has one decimal place (when the channel list includes digital OTA channels).

So for channel 3 you'd enter 3-0 = 3.0, 3.1 you'd enter 3-1 = 3.1...

**Comment by MrChip53** [comment]
I think what i will look at doing is when the user inputs 3 it will have just 3 in the corner but then a drop down list with any .x channels if they exist. That way if a remote doesnt have a . button those channels will still be accessible.

**Comment by MrChip53** [comment]
Closing this for now

**Pull request closed without merging**